File: Java/number-complement.java
class Solution {
    public int findComplement(int num) {
        
        int numberOfBits = numberOfBits(num);
        
        return num ^ ((1 << numberOfBits) - 1);
    }
    
    /**
    * Method to find out the total number of bits in a number.
    **/
    public static int numberOfBits(int num){
        int countOfBits = 0;
        
        while(num != 0){
            num = num >> 1;
            countOfBits++;
        }
        
        return countOfBits;
    }
}

File: C++/Number-Complement.cpp
// Problem URL: https://leetcode.com/problems/single-number-iii/
/*
 * Runtime - 0ms
 * Memory  - 5.8mb
 */

class Solution {
public:
    int findComplement(int num) {
        int inv = 0, iter = 0;
        int reminder;
        while(num) {
            inv = inv | ((1 - (num % 2))<<iter);
            num /= 2;
            iter++;
        }
        return inv;
    }
};

